lib/graphql/query/context.rb

```diff
@@ -155,6 +155,7 @@ def initialize(query:, values: , object:)
         @path = []
         @value = nil
         @context = self # for SharedMethods
+        @scoped_context = {}
```

Comment:
My initial implemented relied on `Context@scoped_context` being an `ImmutableHash` instead of an `Hash`. I ended up opting out of that decision since it provides little benefit unless we can sure that _all_ values, no matter how deeply nested, are immutable. That being said, with `@scoped_context` being a `Hash`, there's nothing stopping an arbitrary resolver from keeping a reference to an object which is used in a separate scoped context, and modifying it from there. I think that's okay though, since doing so would invalidate the goal of keeping this reference only for a field and its children.